### PR TITLE
Overflow menu

### PR DESCRIPTION
**Pull request by MoizMasud**
https://github.com/IBM/carbon-components-builder/issues/46

### CODE REVIEW

**Comment by MoizMasud** [comment]
@zvonimirfras please review and merge

File: src/fragment-components/a-overflow-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+	return (
+		<AComponent
+		componentObj={componentObj}
+		{...rest}>
+				<OverflowMenu
+						flipped={componentObj.flipped}
+						direction={componentObj.placement}
+						className={` ${preventCheckEvent} ${componentObj.cssClasses?.map((cc: any) => cc.id).join(' ')} `}>
+							{
+							componentObj.items.map((step: any, index: number) => (
```
Comment (zvonimirfras) [inline src/fragment-components/a-overflow-menu.tsx:157]: indentation seems off

File: src/fragment-components/a-overflow-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+	...rest
+}: any) => {
+	return (
+		<AComponent
+		componentObj={componentObj}
+		{...rest}>
+				<OverflowMenu
+						flipped={componentObj.flipped}
+						direction={componentObj.placement}
+						className={` ${preventCheckEvent} ${componentObj.cssClasses?.map((cc: any) => cc.id).join(' ')} `}>
```
Comment (zvonimirfras) [inline src/fragment-components/a-overflow-menu.tsx:155]: Should this be `cx` instead? Or some combination?

File: src/fragment-components/a-overflow-menu.tsx [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+							{
+							componentObj.items.map((step: any, index: number) => (
+								<OverflowMenuItem
+									className={step.className}
+									href={step.hasLink ? step.link : undefined}
```
Comment (zvonimirfras) [inline src/fragment-components/a-overflow-menu.tsx:160]: `hasLink` _and_ `link`?

File: src/fragment-components/a-overflow-menu.tsx [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+				className: 'option-2',
+				disabled: false,
+				hasLink: false,
+				isDelete: false,
+				hasDivider: false,
+				link: ''
+			}
+		]
+	},
+	image: image,
```
Comment (zvonimirfras) [inline src/fragment-components/a-overflow-menu.tsx:207]:
```suggestion
	image,
```
?

**Review by zvonimirfras — CHANGES_REQUESTED** [review]

File: src/fragment-components/a-overflow-menu.tsx (hunk 3 above)
Comment (MoizMasud) [reply src/fragment-components/a-overflow-menu.tsx:160]: yes, if the user checks off the hasLink checkbox then we enable the link input and include it in the overflow menu item

Comment (zvonimirfras) [reply src/fragment-components/a-overflow-menu.tsx:163]: We don't have to store hasLink in the componentObj tho, do we? We'll know if it "has link" based on `link` not being undefined.
Checkbox can be a UI thing and when it's unchecked we can clear `link` value. Perhaps after a second or two delay if it's not checked again.

**Review by zvonimirfras — APPROVED** [review]

**Pull request merged**